Documentation/network/bgp-control-plane/bgp-control-plane-v2.rst

```diff
@@ -0,0 +1,765 @@
+
+* ``CiliumBGPClusterConfig``: It defines BGP instances and peer configurations which will be applied to multiple nodes.
+* ``CiliumBGPPeerConfig``: It is common set of BGP peering setting, it can be used across multiple peers.
+* ``CiliumBGPAdvertisements``: It defines prefixes which will be injected into BGP routing table.
+* ``CiliumBGPNodeConfigOverride``: It defines node specific BGP configuration which users may want to fine control.
+
+BGP Cluster Configuration
+=========================
+
+``CiliumBGPClusterConfig`` resource is used to define BGP configuration for one or more nodes in
+the cluster based on its ``nodeSelector`` field. Each ``CiliumBGPClusterConfig`` defines one or
+more BGP instances, which are uniquely identified by ``name`` field.
+
+BGP instance can have one or more peers. Each peer is uniquely identified by ``name`` field. Peer
+autonomous number and peer address are defined by ``peerASN`` and ``peerAddress`` fields,
+respectively. Configuration of the peers is defined by ``peerConfigRef`` field, which is a reference
+to a peer configuration resource. ``Group`` and ``kind`` in ``peerConfigRef`` are optional and defaults to
+``cilium.io`` and ``CiliumBGPPeerConfig`` respectively.
+
+
+.. warning::
+
+    The ``CiliumBGPPeeringPolicy`` and ``CiliumBGPClusterConfig`` should not be used together. If both
+    resources are present and Cilium agent matches with both based on the node selector,
+    ``CiliumBGPPeeringPolicy`` will take precedence.
+
+
+.. code-block:: yaml
+
+    apiVersion: cilium.io/v2alpha1
+    kind: CiliumBGPClusterConfig
+    metadata:
+      name: cilium-bgp
+    spec:
+      nodeSelector:
+        matchLabels:
+          rack: rack0
+      bgpInstances:
+      - name: "instance-65000"
+        localASN: 65000
+        peers:
+        - name: "peer-65000-tor1"
+          peerASN: 65000
+          peerAddress: fd00:10:0:0::1
+          peerConfigRef:
+            name: "cilium-peer"
+        - name: "peer-65000-tor2"
+          peerASN: 65000
+          peerAddress: fd00:11:0:0::1
+          peerConfigRef:
+            name: "cilium-peer"
+
+.. _bgp_peer_configuration:
+
+BGP Peer Configuration
+======================
+
+``CiliumBGPPeerConfig`` resource is used to define BGP peer configuration. Multiple peers can
+share the same configuration and provide reference to the common ``CiliumBGPPeerConfig``
+resource.
+
+It contains configuration options for
+
+- :ref:`Transport <bgp_peer_configuration_transport>`
+- :ref:`MD5 Password <bgp_peer_configuration_password>`
+- :ref:`Timers <bgp_peer_configuration_timers>`
+- :ref:`Graceful Restart <bgp_peer_configuration_graceful_restart>`
+- :ref:`Address Families <bgp_peer_configuration_afi>`
+
+.. code-block:: yaml
+
+    apiVersion: cilium.io/v2alpha1
+    kind: CiliumBGPPeerConfig
+    metadata:
+      name: cilium-peer
+    spec:
+      transport:
+      timers:
+      authSecretRef:
+      gracefulRestart:
+      families:
```

Comment:
Please introduce this snippet as well.

Should the values in the fields be provided?